include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/Solver/IFDSSolver.h

```diff
@@ -37,19 +39,61 @@ class IFDSSolver : public IDESolver<AnalysisDomainExtender<AnalysisDomainTy>> {
   using D = typename AnalysisDomainTy::d_t;
   using N = typename AnalysisDomainTy::n_t;
 
-  IFDSSolver(IFDSTabulationProblem<AnalysisDomainTy> &ifdsProblem)
-      : IDESolver<AnalysisDomainExtender<AnalysisDomainTy>>(ifdsProblem) {}
+  IFDSSolver(IFDSTabulationProblem<AnalysisDomainTy> &IFDSProblem)
+      : IDESolver<AnalysisDomainExtender<AnalysisDomainTy>>(IFDSProblem) {}
 
   ~IFDSSolver() override = default;
 
-  std::set<D> ifdsResultsAt(N stmt) {
+  ///
+  /// Returns the data-flow results at the given statement.
+  ///
+  std::set<D> ifdsResultsAt(N Inst) {
     std::set<D> KeySet;
-    std::unordered_map<D, BinaryDomain> ResultMap = this->resultsAt(stmt);
+    std::unordered_map<D, BinaryDomain> ResultMap = this->resultsAt(Inst);
     for (auto FlowFact : ResultMap) {
       KeySet.insert(FlowFact.first);
     }
     return KeySet;
   }
+
+  ///
+  /// Returns the data-flow results at the given statement while respecting
+  /// LLVM's SSA semantics.
+  ///
+  /// An example: when a value is loaded and the location loaded from, here
+  /// variable 'i', is a data-flow fact that holds, then the loaded value '%0'
+  /// will usually be generated and also holds. However, due to the underlying
+  /// theory (and respective implementation) this load instruction causes the
+  /// loaded value to be generated and thus, it will be valid only AFTER the
+  /// load instruction, i.e., at the successor instruction.
+  ///
+  ///   %0 = load i32, i32* %i, align 4
+  ///
+  /// This result accessor function returns the results at the successor
+  /// instruction(s) reflecting that the expression on the left-hand side holds
+  /// if the expression on the right-hand side holds.
+  ///
+  template <typename NTy = N>
+  typename std::enable_if_t<std::is_same_v<NTy, const llvm::Instruction *>,
+                            std::set<D>>
+  ifdsResultsAtInLLVMSSA(NTy Inst) {
+    std::unordered_map<D, BinaryDomain> ResultMap = [this, Inst]() {
+      if (Inst->getType()->isVoidTy()) {
+        return this->resultsAt(Inst);
+      } else {
+        // In this case we have a value on the left-hand side and must return
+        // the results at the successor instruction. Note that terminator
+        // instructions are always of void type.
+        return this->resultsAt(Inst->getNextNode());
+        assert("Expected to find a successor instruction!");
```

Comment:
```suggestion
        assert(Inst->getNextNode() && "Expected to find a successor instruction!");
        return this->resultsAt(Inst->getNextNode());
```
We should put the assert before the return :)

Reply:
That's a good point actually :+1: